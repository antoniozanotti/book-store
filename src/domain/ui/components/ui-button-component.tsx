import { UiButtonComponentInterface } from "./ui-button-component-interface";

export function UiButtonComponent({
  children,
  ...props
}: UiButtonComponentInterface) {
  const buttonStyles = "bg-primary-900 font-bold p-4 text-primary-100";
  props.className = props.className
    ? `${buttonStyles} ${props.className}`
    : buttonStyles;
    
  return <button {...props}>{children}</button>;
}
