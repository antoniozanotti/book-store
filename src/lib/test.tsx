import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
