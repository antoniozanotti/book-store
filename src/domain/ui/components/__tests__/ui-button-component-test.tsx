
import { render, screen } from "@/lib/test";
import { UiButtonComponent } from "../ui-button-component";

describe("UiButtonComponent", () => {
  it("should render", async () => {
    const { container } = render(<UiButtonComponent>Test</UiButtonComponent>);
    expect(container).toMatchSnapshot();
  });
});