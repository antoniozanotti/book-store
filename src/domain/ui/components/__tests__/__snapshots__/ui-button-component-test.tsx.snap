// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiButtonComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <button
      class="bg-primary-900 font-bold p-4 text-primary-100"
    >
      Test
    </button>
  </div>
</div>
`;
